app/Http/Controllers/QuizController.php

```
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreQuizResultRequest;
+use App\Models\Course;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator;
+
+class QuizController extends Controller
+{
+    public function show(Course $course): View
+    {
+        $quiz = $course->quiz;
+
+        return view('quizzes.show', [
+            'course' => $course,
+            'quiz' => $quiz,
+            'title' => $quiz->title,
+            'questions' => $quiz->questions,
+        ]);
+    }
+
+    public function storeQuizResult(StoreQuizResultRequest $request, Course $course): View | RedirectResponse
+    {
+        $data = $request->validated();
+        $user = Auth::user();
+        $correctQuestions = 0;
+        $quiz = $course->quiz;
+        $quizWithCounts = $quiz->loadCount('questions');
+        $isPass = false;
+        $validator = Validator::make([], []);
+        $previousAnswers = [];
+        $wrongAnswers = [];
+
+        foreach ($quizWithCounts->questions as $question) {
+            $previousAnswers[$question->id] = $data['questions'][$question->id];
+            if ($question->getCorrectChoices() == $data['questions'][$question->id]) {
+                $correctQuestions++;
+            } else {
+                $validator->errors()->add('questions.'.$question->id, __('Wrong answer'));
+                $wrongAnswers[] = $question->id;
+            }
+        }
+
+        //@phpstan-ignore-next-line
+        if ($quizWithCounts->questions_count > 0) {
```

It would seem that this condition should be checked at the top and return immediately if the question_count is 0. For example the foreach wouldn't have any questions to iterate over either. I'm not sure if this should throw an error because in reality this controller function shouldn't be run at all if there are no questions.

---

At the moment what will happen is that the test will go to the results page with a failure result. In other cases of failure that page is redirected back to the quiz with info on incorrect answers.

---

I just removed this check, because we are already checking if there's at least 1 question in the quiz before interacting with this controller method.